wallet/chainntfns.go

```diff
@@ -690,21 +886,24 @@ func (w *Wallet) addRelevantTx(rec *wtxmgr.TxRecord,
 
 	// Send notification of mined or unmined transaction to any interested
 	// clients.
-	//
-	// TODO: Avoid the extra db hits.
-	if block == nil {
-		details, err := w.TxStore.UniqueTxDetails(&rec.Hash, nil)
-		if err != nil {
-			log.Errorf("Cannot query transaction details for notifiation: %v", err)
-		} else {
-			w.NtfnServer.notifyUnminedTransaction(details)
-		}
-	} else {
-		details, err := w.TxStore.UniqueTxDetails(&rec.Hash, &block.Block)
-		if err != nil {
-			log.Errorf("Cannot query transaction details for notifiation: %v", err)
+	if w.NtfnServer != nil && (len(w.NtfnServer.accountClients) > 0 ||
```

Comment:
The calls for UniqueTxDetails are really expensive and if they can be avoided it'd be nice. I didn't think the NtfnServer for wallet were commonly used so I added the check for them.

Is

`(len(w.NtfnServer.accountClients) > 0 || len(w.NtfnServer.spentness) > 0) {`

correct for skipping the ntfn in the event there are no clients? Maybe should be another PR